include/RevThread.h

```diff
@@ -27,37 +27,69 @@ namespace SST::RevCPU{
 
 // Enum for tracking the state of a RevThread.
 enum class ThreadState {
-  START,    // Initial state; resources are allocated.
-  RUNNING,  // Thread is assigned and executing on a Proc/HART.
-  BLOCKED,  // Waiting for I/O or thread synchronization.
+  START,    // Initial state; RevCPU has not yet registered it in its Threads table
+  RUNNING,  // Thread is assigned and executing on a Proc/HART (NOTE: This does NOT change if a Thread is stalled)
+  BLOCKED,  // Waiting for thread synchronization at this point (Currently only triggered by call to `rev_pthread_join`)
   READY,    // In the ready list; waiting for CPU (not implemented yet).
   DONE,     // Thread has finished; deallocate resources.
 };
 
 class RevThread {
+ private:
+  uint32_t ThreadID;                                   // Thread ID
+  uint32_t ParentThreadID;                             // Parent ThreadID
+  uint64_t StackPtr;                                   // Initial stack pointer
+  uint64_t FirstPC;                                    // Initial PC
+  std::shared_ptr<RevMem::MemSegment> ThreadMem;       // TLS and Stack memory
+  RevFeature* Feature;                                 // Feature set for this thread
+  uint64_t ThreadPtr;                                  // Thread pointer
+  ThreadState State = ThreadState::START;              // Thread state (Initializes as )
+  RevRegFile RegFile;                                  // Register file
+  std::unordered_set<uint32_t> ChildrenThreadIDs = {}; // Child thread IDs
+  std::unordered_set<int> fildes = {0, 1, 2  };      // Default file descriptors
+
+  uint32_t WaitingToJoinTID = __INVALID_TID__;
+
  public:
   RevThread(uint32_t inputThreadID, uint32_t inputParentThreadID,
             uint64_t inputStackPtr, uint64_t inputFirstPC,
             std::shared_ptr<RevMem::MemSegment>& inputThreadMem,
-            RevFeature* inputFeature);
+            RevFeature* inputFeature)
+   : ThreadID(inputThreadID), ParentThreadID(inputParentThreadID), StackPtr(inputStackPtr),  
+     FirstPC(inputFirstPC), ThreadMem(inputThreadMem), Feature(inputFeature) {
+  // Create the RegFile for this thread
+  RegFile = RevRegFile(); 
+  
+  // Set the stack pointer 
+  RegFile.SetX(Feature, 2, StackPtr );
+
+  // Set the thread pointer
+  ThreadPtr = ThreadMem->getTopAddr();
+  RegFile.SetX(Feature, 2, ThreadPtr);
```

Comment:
Isn't the thread pointer `X4`?

[Here is the ABI spec](https://github.com/riscv-non-isa/riscv-elf-psabi-doc/releases/download/v1.0/riscv-abi.pdf).